### PR TITLE
Benchmarks: Build Pipeline - Add nccl-tests as a submodule and add build logic.

### PR DESCRIPTION
**Pull request by yukirora**
**Description**
 Add nccl-tests as a submodule and add build logic.

**Major Revision**
- Add nccl-tests as submodule
- Add build script for it.

### CODE REVIEW

**Review by cp5555 — APPROVED** [review]

File: third_party/Makefile [hunk 1]
```diff
@@ -29,3 +30,10 @@ bandwidthTest: sb_micro_path
 	git clone -b v$(shell nvcc --version | grep 'release' | awk '{print $$6}' | cut -c2- | cut -d '.' -f1-2) https://github.com/NVIDIA/cuda-samples.git ./cuda-samples
 	cd ./cuda-samples/Samples/bandwidthTest && make clean && make TARGET_ARCH=x86_64 SMS="70 75 80 86"
 	cp -v ./cuda-samples/Samples/bandwidthTest/bandwidthTest $(SB_MICRO_PATH)/bin/
+# Build nccl-tests.
```
Comment (abuccts) [inline third_party/Makefile:33]: add an empty line between different targets?

File: third_party/Makefile [hunk 2]
```diff
@@ -29,3 +30,10 @@ bandwidthTest: sb_micro_path
...
+# The version we use is commit 44df0bf from master branch, due to it didn't update release tag for long time.
+nccl_tests: sb_micro_path
+ifneq (,$(wildcard nccl-tests/Makefile))
+	cd nccl-tests && make MPI=1 MPI_HOME=/usr/local/mpi/ -j
+	cp nccl-tests/build/* $(SB_MICRO_PATH)/bin/
```
Comment (abuccts) [inline third_party/Makefile:38]: cp -v?

**Review by abuccts — COMMENTED** [review]
pls remove it in dockerfile

File: third_party/Makefile [hunk 3]
```diff
@@ -29,3 +30,10 @@ bandwidthTest: sb_micro_path
 	git clone -b v$(shell nvcc --version | grep 'release' | awk '{print $$6}' | cut -c2- | cut -d '.' -f1-2) https://github.com/NVIDIA/cuda-samples.git ./cuda-samples
 	cd ./cuda-samples/Samples/bandwidthTest && make clean && make TARGET_ARCH=x86_64 SMS="70 75 80 86"
 	cp -v ./cuda-samples/Samples/bandwidthTest/bandwidthTest $(SB_MICRO_PATH)/bin/
+# Build nccl-tests.
+# The version we use is commit 44df0bf from master branch, due to it didn't update release tag for long time.
+nccl_tests: sb_micro_path
+ifneq (,$(wildcard nccl-tests/Makefile))
+	cd nccl-tests && make MPI=1 MPI_HOME=/usr/local/mpi/ -j
```
Comment (guoshzhao) [inline third_party/Makefile:37]: can we confirm the mpi location is always `/usr/local/mpi/`?

File: third_party/Makefile (hunk 1 above)
Comment (yukirora) [reply third_party/Makefile:36]: ok

File: third_party/Makefile (hunk 2 above)
Comment (yukirora) [reply third_party/Makefile]: ok

File: third_party/Makefile (hunk 3 above)
Comment (yukirora) [reply third_party/Makefile]: change to use whereis to find mpi install path

File: third_party/Makefile [hunk 4]
```diff
@@ -29,3 +32,11 @@ bandwidthTest: sb_micro_path
 	git clone -b v$(shell nvcc --version | grep 'release' | awk '{print $$6}' | cut -c2- | cut -d '.' -f1-2) https://github.com/NVIDIA/cuda-samples.git ./cuda-samples
 	cd ./cuda-samples/Samples/bandwidthTest && make clean && make TARGET_ARCH=x86_64 SMS="70 75 80 86"
 	cp -v ./cuda-samples/Samples/bandwidthTest/bandwidthTest $(SB_MICRO_PATH)/bin/
+
+# Build nccl-tests.
+# The version we use is commit 44df0bf from master branch, due to it didn't update release tag for long time.
```
Comment (guoshzhao) [inline third_party/Makefile:37]: due to => since?

**Review by guoshzhao — APPROVED** [review]

File: third_party/Makefile [hunk 5]
```diff
@@ -29,3 +32,11 @@ bandwidthTest: sb_micro_path
...
+nccl_tests: sb_micro_path
+ifneq (,$(wildcard nccl-tests/Makefile))
+	cd ./nccl-tests && make MPI=1 MPI_HOME=$(shell whereis mpi | grep 'mpi' | awk '{print $$2}') -j
```
Comment (abuccts) [inline third_party/Makefile:40]: there are multiple mpi in the image, pls make it configurable, e.g., add `MPI_HOME ?= /usr/local/mpi` line at the beginning

File: third_party/Makefile (hunk 4 above)
Comment (yukirora) [reply third_party/Makefile]: changed

File: third_party/Makefile (hunk 3 above)
Comment (yukirora) [reply third_party/Makefile]: Yifan suggests to make it a variable

**Review by abuccts — APPROVED** [review]

File: third_party/Makefile (hunk 3 above)
Comment (guoshzhao) [reply third_party/Makefile]: OK

**Pull request merged**